src/global/signal_handler.cc

```diff
@@ -163,18 +163,21 @@ void get_name_by_pid(pid_t pid, char *task_name)
     derr << "Fail to open '" << proc_pid_path 
          << "' error = " << cpp_strerror(fd) 
          << dendl;
-    return;
+    return "<unknown>";
   }
 
-  int ret = read(fd, task_name, PATH_MAX-1);
+  char buf[PATH_MAX] = {0};
+  int ret = read(fd, buf, sizeof(buf));
   if (ret < 0) {
     ret = -errno;
-    derr << "Fail to read '" << proc_pid_path 
-         << "' error = " << cpp_strerror(ret) 
+    derr << "Fail to read '" << proc_pid_path
+         << "' error = " << cpp_strerror(ret)
          << dendl;
+    return "<unknown>";
```

Comment:
I think you may need an extra
  close(fd);
before return, otherwise there is a leak.